CMake error during the build on Fedora 38

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.


### Version

2.6.0-rc2

### Minimal reproduce step

Just run the `cmake` command.

### What did you expect to see?

Successful build.

### What did you see instead?

```
Fetching jsoncons...
CMake Deprecation Warning at /home/yaroslav/tests/kvrocks/2.6.0-rc/build/_deps/jsoncons-src/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- jsoncons v0.171.0
CMake Error at cmake/xxhash.cmake:34 (add_subdirectory):
  add_subdirectory not given a binary directory but the given source
  directory
  "/home/yaroslav/tests/kvrocks/2.6.0-rc/build/_deps/xxhash-src/cmake_unofficial"
  is not a subdirectory of
  "/home/yaroslav/tests/kvrocks/2.6.0-rc/apache-kvrocks-2.6.0-src".  When
  specifying an out-of-tree source a binary directory must be explicitly
  specified.
Call Stack (most recent call first):
  CMakeLists.txt:134 (include)


Fetching span...
-- Project 'span_lite', package 'span-lite' version: '0.10.3'
-- Configuring incomplete, errors occurred!
```


### Anything Else?

On my machine:

```
[yaroslav@localhost build]$ cmake --version
cmake version 3.27.4
```


### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/cmake/xxhash.cmake
+++ b/cmake/xxhash.cmake
@@ -31,5 +31,5 @@ if(NOT xxhash_POPULATED)
   set(BUILD_SHARED_LIBS OFF)
   set(XXHASH_BUILD_XXHSUM OFF)
 
-  add_subdirectory(${xxhash_SOURCE_DIR}/cmake_unofficial ${xxhash_BUILD_DIR} EXCLUDE_FROM_ALL)
+  add_subdirectory(${xxhash_SOURCE_DIR}/cmake_unofficial ${xxhash_BINARY_DIR} EXCLUDE_FROM_ALL)
 endif()

